numpy/core/src/umath/loops_arithmetic.dispatch.c.src

```diff
@@ -101,6 +101,82 @@ simd_divide_by_scalar_contig_@sfx@(char **args, npy_intp len)
 }
 /**end repeat**/
 
+#if defined(NPY_HAVE_VSX4)
+/*
+ * As Power10 only supports integer vector division for data of 32 bits or
+ * greater, we have to convert npyv_u8 into 4x npyv_u32, execute the integer
+ * vector division instruction, and then, convert the result back to npyv_u8.
+ */
+NPY_FINLINE npyv_u8 vsx4_div_u8(npyv_u8 a, npyv_u8 b)
+{
+    npyv_u16x2 a_expand = npyv_expand_u16_u8(a);
+    npyv_u32x2 ahi = npyv_expand_u32_u16(a_expand.val[0]);
+    npyv_u32x2 alo = npyv_expand_u32_u16(a_expand.val[1]);
+    npyv_u16x2 b_expand = npyv_expand_u16_u8(b);
+    npyv_u32x2 bhi = npyv_expand_u32_u16(b_expand.val[0]);
+    npyv_u32x2 blo = npyv_expand_u32_u16(b_expand.val[1]);
+    npyv_u32 divhi1 = vec_div(ahi.val[0], bhi.val[0]);
+    npyv_u32 divlo1 = vec_div(ahi.val[1], bhi.val[1]);
+    npyv_u32 divhi2 = vec_div(alo.val[0], blo.val[0]);
+    npyv_u32 divlo2 = vec_div(alo.val[1], blo.val[1]);
+    npyv_u16 reshi = (npyv_u16)vec_pack(divhi1, divlo1);
+    npyv_u16 reslo = (npyv_u16)vec_pack(divhi2, divlo2);
+    npyv_u8 res = (npyv_u8)vec_pack(reshi, reslo);
+    return res;
+}
+
+NPY_FINLINE npyv_u16 vsx4_div_u16(npyv_u16 a, npyv_u16 b)
+{
+    npyv_u32x2 a_expand = npyv_expand_u32_u16(a);
+    npyv_u32x2 b_expand = npyv_expand_u32_u16(b);
+    npyv_u32 divhi = vec_div(a_expand.val[0], b_expand.val[0]);
+    npyv_u32 divlo = vec_div(a_expand.val[1], b_expand.val[1]);
+    npyv_u16 res = (npyv_u16)vec_pack(divhi, divlo);
+    return res;
+}
+
+#define vsx4_div_u32 vec_div
+
+/**begin repeat
+ * Unsigned types
+ * #sfx    = u8, u16, u32#
+ * #len    = 8,  16,  32#
+ */
+static NPY_INLINE void
+vsx4_simd_divide_contig_@sfx@(char **args, npy_intp len)
+{
+    npyv_lanetype_@sfx@ *src1  = (npyv_lanetype_@sfx@ *) args[0];
+    npyv_lanetype_@sfx@ *src2  = (npyv_lanetype_@sfx@ *) args[1];
+    npyv_lanetype_@sfx@ *dst   = (npyv_lanetype_@sfx@ *) args[2];
+    const int vstep            = npyv_nlanes_@sfx@;
+    const npyv_@sfx@ zero      = npyv_zero_@sfx@();
+
+    for (; len >= vstep; len -= vstep, src1 += vstep, src2 += vstep,
+         dst += vstep) {
+        npyv_@sfx@ a = npyv_load_@sfx@(src1);
+        npyv_@sfx@ b = npyv_load_@sfx@(src2);
+        npyv_@sfx@ c = vsx4_div_@sfx@(a, b);
+        npyv_store_@sfx@(dst, c);
+        if (vec_any_eq(b, zero)) {
+            npy_set_floatstatus_divbyzero();
+        }
+    }
```

Comment:
```suggestion
    npyv_@bsfx@ is_bzero = npyv_cvt_@bsfx@_@sfx@(zero);
    for (; len >= vstep; len -= vstep, src1 += vstep, src2 += vstep,
         dst += vstep) {
        npyv_@sfx@ a = npyv_load_@sfx@(src1);
        npyv_@sfx@ b = npyv_load_@sfx@(src2);
        npyv_@sfx@ c = vsx4_div_@sfx@(a, b);
        is_bzero = npyv_or_@bsfx@(is_bzero, npyv_cmpeq_@sfx@(b, zero));
        npyv_store_@sfx@(dst, c);
    }
    if (vec_any_eq(b, zero)) {
        npy_set_floatstatus_divbyzero();
    }
```
reduce jmps should increase performance. also `equal` and `or` should be faster `vec_any` isn't? if not then wrap `vec_any_eq()` by `NPY_UNLIKELY()`.

Reply:
Hi @seiko2plus, thanks for the review! 

I tried to remove the control flow following your suggestion (and also using some other mechanisms), but neither one improved the execution time. For some cases, I see a slowdown between ~5-7%. As it didn't improve, I only added `NPY_UNLIKELY()`.